lib/Model/Pass/PurgeUnnamedAndUnreachableTypes.cpp

```diff
@@ -20,10 +26,74 @@ static RegisterModelPass R("purge-unnamed-and-unreachable-types",
 
 static RegisterModelPass R2("purge-unreachable-types",
                             "Remove all the types that cannot be reached from "
-                            "a type \"outside\" the type "
-                            "system itself",
+                            "a type \"outside\" the type system itself",
                             model::purgeUnreachableTypes);
 
+// Helper for fixing array constness.
+static void fixConstArrays(model::Type &Type) {
+  if (auto *Array = llvm::dyn_cast<model::ArrayType>(&Type)) {
+    if (Array->IsConst()) {
+      Array->ElementType()->IsConst() = true;
+      Array->IsConst() = false;
+    }
+
+    if (not Array->ElementType().empty())
+      fixConstArrays(*Array->ElementType());
+
+  } else if (auto *Pointer = llvm::dyn_cast<model::PointerType>(&Type)) {
+    if (not Pointer->PointeeType().empty())
+      fixConstArrays(*Pointer->PointeeType());
+
+  } else {
+    // Here we assume that this helper is run for every definition, as such
+    // there's no need to unwrap `model::DefinedType`s.
+    revng_assert(llvm::isa<model::DefinedType>(Type)
+                 || llvm::isa<model::PrimitiveType>(Type));
+  }
+}
+
+static Logger<> PurgeInvalidLogger("purge-invalid-types");
```

Comment:
I'd tend to have a single logger for the whole `.cpp` file, move it to the top and call it `Log`.

Reply:
Done!